CMakeLists.txt

```diff
@@ -229,6 +229,13 @@ add_custom_target(
     DEPENDS ZAPD
 )
 
+add_custom_target(
+    CompileLogic
```

Comment:
not sure how best to wire this with CMake, but also how do we want to manage this? should generated.cpp be checked in or should it be left to the build process? leaning towards the latter